<launch>

  <arg name="UAV_NAME" default="$(optenv UAV_NAME)"/>
  <arg name="NATO_NAME" default="$(optenv NATO_NAME)"/>
  <arg name="UAV_TYPE" default="$(optenv UAV_TYPE)"/>
  <arg name="UAV_MASS" default="$(optenv UAV_MASS)"/>
  <arg name="SENSORS" default="$(optenv SENSORS)"/>
  <arg name="RUN_TYPE" default="$(optenv RUN_TYPE)"/>
  <arg name="PIXGARM" default="$(optenv PIXGARM false)"/>
  <arg name="COLORSCHEME" default="$(optenv PROFILES_BOTH COLORSCHEME_LIGHT)"/>
  <arg name="RAINBOW" default="false"/>
	<arg name="OLD_PX4_FW" default="$(optenv OLD_PX4_FW false)" />

  <arg name="debug" default="false"/>

  <arg     if="$(arg debug)" name="launch_prefix" value="debug_roslaunch" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />

  <arg name="config_file" default=""/>

  <arg if="$(eval arg('RUN_TYPE') == 'realworld')" name="UAV_MASS" default="$(arg UAV_MASS)" />

  <group ns="$(arg UAV_NAME)">

    <node name="mrs_uav_status_acquisition" pkg="mrs_uav_status" type="data_acquisition" output="screen" respawn="true" launch-prefix="$(arg launch_prefix)">

      <param name="uav_name" value="$(arg UAV_NAME)" />
      <param name="nato_name" value="$(arg NATO_NAME)" />
      <param name="uav_type" value="$(arg UAV_TYPE)" />
      <param name="run_type" value="$(arg RUN_TYPE)" />
      <param name="sensors" value="$(arg SENSORS)" />
      <param name="pixgarm" type="bool" value="$(arg PIXGARM)" />

      <param if="$(eval arg('RUN_TYPE') == 'realworld')" name="uav_mass" value="$(arg UAV_MASS)" />

      <!-- TODO this needs to be somehow obtained from the control manager -->
      <param if="$(eval arg('RUN_TYPE') == 'simulation')" name="uav_mass" value="1.0" />

      <rosparam file="$(find mrs_uav_status)/config/default.yaml" command="load" />
      <rosparam if="$(eval not arg('config_file') == '')" file="$(arg config_file)" />

        <!-- Subscribers -->
      <remap from="~uav_state_in" to="estimation_manager/uav_state" />
      <remap from="~cmd_tracker_in" to="control_manager/tracker_cmd" />
      <remap from="~estimation_diag_in" to="estimation_manager/diagnostics" />
      <remap from="~mpc_diag_in" to="control_manager/mpc_tracker/diagnostics" />
      <remap from="~hw_api_status_in" to="hw_api/status" />
      <remap from="~cmd_attitude_in" to="control_manager/attitude_cmd" />
      <remap from="~battery_in" to="hw_api/battery_state" />
      <remap from="~control_manager_in" to="control_manager/diagnostics" />
      <remap from="~gain_manager_in" to="gain_manager/diagnostics" />
      <remap from="~constraint_manager_in" to="constraint_manager/diagnostics" />
      <remap from="~tf_static_in" to="/tf_static" />
      <remap from="~gnss_in" to="hw_api/gnss" />
      <remap from="~gnss_status_in" to="hw_api/gnss_status" />
      <remap from="~odometry_in" to="hw_api/odometry" />
      <remap from="~automatic_start_in" to="automatic_start/can_takeoff" />
      <remap from="~throttle_in" to="control_manager/throttle" />
      <remap from="~mass_estimate_in" to="control_manager/mass_estimate" />
      <remap from="~mass_set_in" to="TODO_PROVIDE_MASS_SET_TOPIC" />
      <remap from="~mag_in" to="hw_api/magnetic_field" />
      <remap from="~string_in" to="mrs_uav_status/display_string" />

        <!-- Publishers -->
      <remap from="~uav_status_out" to="mrs_uav_status/uav_status" />
      <remap from="~uav_status_short_out" to="mrs_uav_status/uav_status_short" />
      <remap from="~profiler" to="profiler" />


    </node>

  </group>

</launch>
